File: __tests__/affirmationData.test.ts
import { affirmationCategories } from '../src/data/affirmationData';

describe('Affirmation Data', () => {
  test('should have exactly 12 categories', () => {
    expect(affirmationCategories.length).toBe(12);
  });

  test('all categories should have names and descriptions', () => {
    affirmationCategories.forEach(category => {
      expect(category.name).toBeTruthy();
      expect(category.description).toBeTruthy();
      expect(category.affirmations.length).toBeGreaterThan(0);
    });
  });

  test('should contain expected category names', () => {
    const categoryNames = affirmationCategories.map(c => c.name);
    const expectedCategories = [
      'Self-Love',
      'Confidence', 
      'Success',
      'Health',
      'Abundance',
      'Gratitude',
      'Relationships',
      'Healing',
      'Mindfulness',
      'Motivation',
      'Positivity',
      'Stress Relief'
    ];

    expectedCategories.forEach(expected => {
      expect(categoryNames).toContain(expected);
    });
  });

  test('each category should have at least 8 affirmations', () => {
    affirmationCategories.forEach(category => {
      expect(category.affirmations.length).toBeGreaterThanOrEqual(8);
    });
  });
});

File: test/affirmation_data_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:affirmation_app/models/affirmation_data.dart';

void main() {
  group('Affirmation Data Tests', () {
    test('should have exactly 12 categories', () {
      expect(AffirmationData.categories.length, equals(12));
    });

    test('all categories should have names and descriptions', () {
      for (final category in AffirmationData.categories) {
        expect(category.name.isNotEmpty, isTrue);
        expect(category.description.isNotEmpty, isTrue);
        expect(category.affirmations.isNotEmpty, isTrue);
      }
    });

    test('should contain expected category names', () {
      final categoryNames = AffirmationData.categories.map((c) => c.name).toList();
      final expectedCategories = [
        'Self-Love',
        'Confidence', 
        'Success',
        'Health',
        'Abundance',
        'Gratitude',
        'Relationships',
        'Healing',
        'Mindfulness',
        'Motivation',
        'Positivity',
        'Stress Relief'
      ];
      
      for (final expected in expectedCategories) {
        expect(categoryNames.contains(expected), isTrue, 
            reason: 'Category "$expected" should be present');
      }
    });
  });
}